### DISCUSSION TITLE
SplashScreen setup when using Microsofts Template that implements the use of an Activation Service

### DISCUSSION THREAD

**Discussion by JoeTomkinson** [dotMorten/WinUIEx#83] [Q&A]
It's great to see that WinUIEx seems to be included now as part of the Microsofts Template builder for WinUI3 apps, congrats on that by the way.

I had a question however related to the example readme for your splash screen implementation in regards to the way that Microsofts template build activates the main window. For example:

In App.cs

```
protected async override void OnLaunched(LaunchActivatedEventArgs args)
{
    base.OnLaunched(args);
    await App.GetService<IActivationService>().ActivateAsync(args);
}
```

Which calls activate async in the activation service class:

```
public async Task ActivateAsync(object activationArgs)
{
    // Execute tasks before activation.
    await InitializeAsync();

    // Set the MainWindow Content.
    if (App.MainWindow.Content == null)
    {
        _shell = App.GetService<ShellPage>();
        App.MainWindow.Content = _shell ?? new Frame();
    }

    // Handle activation via ActivationHandlers.
    await HandleActivationAsync(activationArgs);

    // Activate the MainWindow.
    App.MainWindow.Activate();

    // Execute tasks after activation.
    await StartupAsync();
}
```

`App.MainWindow` is:

```
public static WindowEx MainWindow { get; } = new MainWindow();
```

So I was just wondering how I should work in the Splashscreen when the activation pipeline is done this way.

**Comment by dotMorten** [comment]
I’m not super familiar with the template studio architecture. Perhaps @mikebattista would know this better?

**Comment by mikebattista** [comment]
Looks like you need to activate the splash screen immediately as in the sample. Your long-running task is the content of ActivateAsync, so rearrange the code so the SplashScreen knows to block on it. Instead of updating m_window in the callback, update App.MainWindow.

**Comment by JoeTomkinson** [comment]
Is there any way to not have to provide the MainWindow to the SplashScreen and handle the activation of the main window and its content using the activation service? I can see that the Splashscreen class doesn't seem to let you do that yourself.

Just curious as running it the way you suggested, at least in my example, seems to generate a corrupted memory exception.

```
System.AccessViolationException: 'Attempted to read or write protected memory. 
This is often an indication that other memory is corrupt.'
```

**Reply by dotMorten** [reply]
Currently not. Feel free to log an issue and I’ll try and look into improving this. (I’m in the process of moving though so spare time is currently limited to work on new features)
